Docs: library panel (#35272)
grafana/grafana: docs/sources/panels/panel-library.md

+++

> title = "Library panel"

Reviewer: I suggest making this and the title plural.

Author: Done

weight = 400
+++

# Library panel

> Library panels allow users to create reusable panels that aren’t tied to any one dashboard. As a result, any changes made to one instance of the library panel affecting all other instances.

Reviewer: We need to more fully explain what library panels are. This can be done later.

Author: Agree

## Manage library panels

To access existing library panels:

> 1. Hover over the **Dashboard** option on the left menu, then select **Library panels** from the drop down options. You can see a list of previusly defined library panels.

Reviewer: ```suggestion
1. Hover over the **Dashboard** option on the left menu, then click **Library panels**. You can see a list of previously defined library panels.
```

Author: Done

1. In **Library panel name**, enter the name.
1. In **Save in folder**, select the folder to save the library panel. By default, the General folder is selected.
1. Click **Create library panel** to save your changes.

Once created, you can modify the library panel.  can be modified, and once saved, all instances of the library panel will represent these modifications.

## Add a library panel

1. Hover over the **+** option on the left menu, then select **Create** from the drop down options. The Add panel dialog opens.
   {{< figure src="/static/img/docs/library-panels/add-library-panel-8-0.png" class="docs-image--no-shadow" max-width= "900px" caption="Screenshot of the edit panel" >}}
1. Click the **Add a panel from the panel library** option. You will see a list of your library panels. 
1. Filter the list or search to find the panel you want to add.
1. Click your desired panel and add it to the dashboard.

## Unlink a library panel

In case you have a library panel on your dashboard that want to modify it without affecting all other instances of the library panel, you can unlink the library panel. 

To unlink a library panel:

> 1. Hover over **Dashboard** on the left menu, and select Library panels from the drop down options.

Reviewer: ```suggestion
1. Hover over **Dashboard** on the left menu, and then click **Library panels**.
```

+++
title = "Library panels"
weight = 400
+++

# Library panel

> Library panels allow users to create reusable panels that aren’t tied to any one dashboard. As a result, any changes made to one instance of the library panel affecting all other instances.

Reviewer 2: Suggestion from the word doc. My goal was to reframe the value statement from what they aren't - tied to one dashboard - to the value they offer: reuse:

```Library panels allow users to create reusable panels where any changes made to one instance of the library panel is reflected on every dashboard affecting all other instances. where the panel is used. These panels can be saved in folders alongside Dashboards and streamline reuse of panels across multiple dashboards.
```

Author: Added, thank you!

## Create a library panel

Before you can use library panels, you have to create them.

To create a library panel:

> 1. Create a panel as you normally would, following instructions in [Add a panel]({{< relref "./add-a-panel.md" >}}).

Reviewer 2: How about:

1. Pick an existing panel or create a panel as you normally would, following instructions in [Add a panel]({{< relref "./add-a-panel.md" >}}).

I think that for many of our community members, they'll be approaching this from an existing panel perspective.

Author: Add a panel is really creating a panel. I will modify this a little - to make it clear that you can use an existing panel as well.

across multiple dashboards.

## Create a library panel

Before you can use library panels, you have to create them.

To create a library panel:

1. Create a Grafana panel as you normally would, following instructions in [Add a panel]({{< relref "./add-a-panel.md" >}}). You an also use an existing panel.
1. Click the title of the panel and then click **Edit**. The panel opens in edit mode.
1. In the panel display options (side pane), click the down arrow option to bring changes to the visualization.
   {{< figure src="/static/img/docs/library-panels/create-lib-panel-from-edit-8-0.png" class="docs-image--no-shadow" max-width= "900px" caption="Screenshot of the edit panel" >}}
1. Click the **Library panels** option, and then click **Create new library panel** to open the create dialog.
   {{< figure src="/static/img/docs/library-panels/create-lib-panel-8-0.png" class="docs-image--no-shadow" max-width= "900px" caption="Screenshot of the create library panel dialog" >}}
1. In **Library panel name**, enter the name.
1. In **Save in folder**, select the folder to save the library panel. By default, the General folder is selected.

> 1. Click **Create library panel** to save your changes.

Reviewer 2: One more line item at the end:

`1. Save the dashboard.`

(This one catches me out all the time as I don't expect to have to save the dashboard and sometimes I'm not prompted to.)

Author: Added.

Optionally, you can click title of the panel and then click **More > Create Library panel**. 

   {{< figure src="/static/img/docs/library-panels/create-lib-panel-from-edit-8-0.png" class="docs-image--no-shadow" max-width= "900px" caption="Screenshot of the edit panel" >}}

>    Then follow the instructions from 

Reviewer 2: I think this is just a vestige from an earlier version...

> Once created, you can modify the library panel.  can be modified, and once saved, all instances of the library panel will represent these modifications.

Reviewer 2: Suggestion: 

**Once created, you can modify the library panel using any dashboard on which it appears. Once the library panel changes are saved, all instances of the library panel will reflect these modifications.**

Author: Added.

   {{< figure src="/static/img/docs/library-panels/library-panel-list-8-0.png" class="docs-image--no-shadow" max-width= "900px" caption="Screenshot of the edit panel" >}}
1. Search for a specific library panel if you know its name. You can also filter the panels by folder or type.

## Unlink a library panel

In case you have a library panel on your dashboard that want to modify it without affecting all other instances of the library panel, you can unlink the library panel. 

To unlink a library panel from a dashboard:

1. Hover over **Dashboard** on the left menu, and then click **Library panels**.
1. Select a library panel that is being used in different dashboards. You will see a list of all the dashboards where the library panel is used.
   {{< figure src="/static/img/docs/library-panels/used-in-list-8-0.png" class="docs-image--no-shadow" max-width= "900px" caption="Screenshot of the edit panel" >}}
1. Select the panel you want to unlink and update. 
1. Click the title of the panel and then click **Edit**. The panel opens in edit mode.
1. Click the **Unlink** option on the top right corner of the UI.

## Delete a library panel

> 

Reviewer 2: Suggestion to add:

**In order to delete a library panel, it must no longer be in use on any dashboard.**

Author: Added
